src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs

```diff
@@ -381,6 +320,19 @@ public static bool IsSpecialRenderPackage(DependencyObject element)
         //#endregion
 
         #region utils
+        
+        /// <summary>
+        /// Sets the property value on the DynamoMeshCore - this makes its way down to our shader and invalidates the render.
+        /// </summary>
+        /// <param name="meshGeom"></param>
+        /// <param name="args"></param>
+        private static void HandleMeshPropertyChange(DynamoGeometryModel3D meshGeom, DependencyPropertyChangedEventArgs args)
+        {
+            if ((meshGeom.SceneNode.RenderCore as DynamoGeometryMeshCore) != null)
+            {
+                (meshGeom.SceneNode.RenderCore as DynamoGeometryMeshCore).SetPropertyData(args);
```

Comment:
this method directs dependency property changes on the GeometryModels and updates the meshCore class - this has the effect of updating the shader.